<launch>
    <arg name="vehicle" default="auk"/>
    <arg name="vehicle_type" default="quad"/>
    <arg name="run_estimator" default="true"/>

    <arg name="record_bag" default="true"/>
    <arg name="bag_name" default="position_hold"/>


    <!--    Launches the ROSFlight node to interface with the Flip32 Board-->
    <node name="rosflight" pkg="rosflight" type="rosflight_io" output="screen">
        <param name="port" value="/dev/ttyUSB0"/>
    </node>

    <!--    Runs the ROS VRPN Node to interface with the mocap system-->
    <node pkg="ros_vrpn_client" name="$(arg vehicle)" type="ros_vrpn_client" args="_vrpn_server_ip:=192.168.1.104" required="true" output="screen">
        <remap from="$(arg vehicle)/ned/pose_stamped" to="pose_stamped"/>
    </node>

    <!--    Loads the camera to mocap-marker calibration-->
    <rosparam command="load" file="$(find reef_estimator)/params/$(arg vehicle)_camera.yaml" />

    <group if="$(arg run_estimator)">
        <node name="reef_estimator" pkg="reef_estimator" type="reef_estimator" output="screen">
            <rosparam file="$(find reef_estimator)/params/xy_est_params.yaml" />
            <rosparam file="$(find reef_estimator)/params/z_est_params.yaml" />
            <rosparam file="$(find reef_estimator)/params/basic_params.yaml" />
            <rosparam>
                enable_rgbd: false
                enable_sonar: false
                enable_mocap_xy: true
                enable_mocap_z: true
            </rosparam>
            <remap from="mocap_ned" to="pose_stamped"/>
        </node>

        <node name="delta_odom" pkg="delta_odom" type="node" output="screen" >
            <rosparam command="load" file="$(find delta_odom)/params/basic.yaml" />
        </node>
    </group>

    <!--    Runs the REEF Control node-->
    <rosparam file="$(find reef_control)/params/$(arg vehicle_type)_pid.yaml" />
    <node name="reef_control_pid" pkg="reef_control" type="reef_control_node" output="screen">
	<remap from="pose_stamped" to="xyz_pose"/>
    </node>

    <arg name="waypoints_file" default="$(find setpoint_generator)/waypoint_files/basic_waypoints.yaml" />
    <node name="setpoint_publisher" pkg="setpoint_generator" type="get_setpoint_node.py" clear_params="true" output="screen">
        <rosparam file="$(arg waypoints_file)"/>
        <rosparam file="$(find setpoint_generator)/params/basic_param.yaml"/>
    </node>

    <node if="$(arg record_bag)" name="record" pkg="rosbag" type="record" args="-O $(arg bag_name) cam_to_init attitude attitude/euler rc_raw imu/data sonar status pose_stamped xyz_debug_estimate xyz_pose xyz_estimate velocity/body_level_frame controller_state xyz_debug_estimate rgbd_velocity/body_level_frame"/>
</launch>
